crates/chain/src/chain_data.rs

```diff
@@ -215,4 +241,64 @@ impl<I: ChainPosition> FullTxOut<I> {
     }
 }
 
+impl<A: Anchor> FullTxOut<ObservedAs<A>> {
+    /// Whether the `txout` is considered mature.
+    ///
+    /// This is the alternative version of [`is_mature`] which depends on `chain_position` being a
+    /// [`ObservedAs<A>`] where `A` implements [`Anchor`].
+    ///
+    /// [`is_mature`]: Self::is_mature
+    pub fn is_mature(&self, tip: u32) -> bool {
+        if !self.is_on_coinbase {
+            return true;
+        }
```

Comment:
This seems like a odd behavior to me. For non-coinbase txs, we will always say it is "mature" even if it's an unconfirmed tx? We should at least filter by unconfirmed, for non-coinbase txs.

Reply:
I think this is out of scope for this PR. This is the same behavior as `is_mature()` with `P: ChainPosition` generic bounds.

Reply:
I don't think this is out of scope. Getting an unconfirmed tx as "mature" doesn't seem logical. Just add one more filter and i's good.

Reply:
It's out of scope because I literally copied the method here:

https://github.com/bitcoindevkit/bdk/blob/165b874dfedb2e05a7ea5d56e6f80577ada48d73/crates/chain/src/chain_data.rs#L225-L242

Reply:
Yes I saw that it's the same as `is_mature`. My point was, as this PR is already big enough, won't hurt to include one-liner fixes. And both of them need it. But agreed, can be done in a separate PR.